### ISSUE TITLE
Issue with password hash version on Encryptor

### ISSUE THREAD

**Issue by ianitsky** [magento/magento2#21289]
### Summary (*)

Maybe this issue are visible only for me because I some things in the password section in the core module.
But I experienced one error that don't do sense for me.

This is the error message:
```
Fatal Error: 'Uncaught TypeError: explode() expects parameter 2 to be string, integer given in /var/www/Magento/vendor/magento/framework/Encryption/Encryptor.php:274 
Stack trace:
#0 /var/www/Magento/vendor/magento/framework/Encryption/Encryptor.php(274):  (':', 1) 
#1 /var/www/Magento/vendor/magento/framework/Encryption/Encryptor.php(223): Magento\Framework\Encryption\Encryptor->getPasswordVersion() 
#2 /var/www/Magento/vendor/magento/module-user/Observer/Backend/AuthObserver.php(145): Magento\Framework\Encryption\Encryptor->validateHashVersion('$2y$13$gkz3jnnL...', true) 
#3 /var/www/Magento/vendor/magento/framework/Event/Invoker/InvokerDefault.php(72): Magento\User\Observer\Backend\AuthObserver->execute(Object(Magento\Framework\Event\Observer)) 
#4 /var/www/Magento/vendor/magento/framework/Event/Invoker/InvokerDefault.php(60): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod(Object(Magento\User\Observer\Backend\AuthObserver), Object(Magento\Framework\Event\Observer)) 
#5 ' in '/var/www/Magento/vendor/magento/framework/Encryption/Encryptor.php' on line 274
```
On the file: Magento\Framework\Encryption\Encryptor:274 we have the explode function:
```php
return array_map('intval', explode(self::DELIMITER, $this->passwordHashMap[self::PASSWORD_VERSION]));
```
But the value of $this->passwordHashMap is:
```php
//Line 83
/**
 * @var array map of password hash
 */
private $passwordHashMap = [
    self::PASSWORD_HASH => '',
    self::PASSWORD_SALT => '',
    self::PASSWORD_VERSION => self::HASH_VERSION_LATEST
];
```
after evaluate:
```
/**
 * @var array map of password hash
 */
private $passwordHashMap = [
    0 => '',
    1 => '',
    2 => 1
];
```
because the value of the constants are:
```php
//Line 33
/**
 * Key of latest used algorithm
 */
const HASH_VERSION_LATEST = 1;

//Line 43
/**#@+
 * Exploded password hash keys
 */
const PASSWORD_HASH = 0;
const PASSWORD_SALT = 1;
const PASSWORD_VERSION = 2;
```

Don't have how explode on line 274 contains a string, only the integer 1.

I'm right?
### Proposed solution

I'm dont have one suggest right now.

**Comment by magento-engcom-team** [comment]
Hi @ianitsky. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@ianitsky do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @ianitsky thank you for you report, this already fixed in 2.3-develop branch 
![deepinscreenshot_select-area_20190218123530](https://user-images.githubusercontent.com/40756439/52945150-beffb800-3379-11e9-8ff1-dec36eb8f748.png)

**Comment by vero2k2** [comment]
hi, 
I have the same error,can you tell me that I modify please

Thank you
